iceoryx_posh/BUILD.bazel

```diff
@@ -25,18 +26,57 @@ configure_file(
     name = "iceoryx_posh_deployment_hpp",
     src = "cmake/iceoryx_posh_deployment.hpp.in",
     out = "include/iceoryx_posh/iceoryx_posh_deployment.hpp",
-    config = {
+    config = select({
         # FIXME: for values see "iceoryx_posh/cmake/IceoryxPoshDeployment.cmake" ... for now some nice defaults
-        "IOX_COMMUNICATION_POLICY": "ManyToManyPolicy",
-        "IOX_INTERNAL_MAX_NUMBER_OF_NOTIFIERS": "256",
-        "IOX_MAX_CHUNKS_ALLOCATED_PER_PUBLISHER_SIMULTANEOUSLY": "8",
-        "IOX_MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY": "256",
-        "IOX_MAX_INTERFACE_NUMBER": "4",
-        "IOX_MAX_PUBLISHERS": "512",
-        "IOX_MAX_PUBLISHER_HISTORY": "16",
-        "IOX_MAX_SUBSCRIBERS": "1024",
-        "IOX_MAX_SUBSCRIBERS_PER_PUBLISHER": "256",
-    },
+        "@platforms//os:macos": {
```

Comment:
I am totally ignorant on Bazel, so I dont know how to do this. We just need to set the value of `IOX_MAX_RUNTIME_NAME_LENGTH` to 98 on Mac and on 100 on Linux, like in CMake. There must be a more elegant way, but this might just work.

Reply:
@JakubSosnovec I think we can leave it for now and live with the code duplication here, the issue seems to be that we cannot concatenate dicts with dicts coming from `select` statements properly.
Related issue: https://github.com/bazelbuild/bazel/issues/12457